### ISSUE TITLE
question about the training or critic network for DDPG

### ISSUE THREAD

**Issue by linyufeishi** [MorvanZhou/Reinforcement-learning-with-tensorflow#109]
question about the training of critic network for DDPG
hello,morvan,fisrtly thanks for your code , I have doubts about the training of the critic network. the function optimizer.minimize(self.loss,global_step =self.global_step) is used to calculate gradients of trainable variables and then update them，I print tf.trainable_variables before it and find that there are some trainable variables of eval_net of actor .so I want to ask that if we directly use the minimize function, the trainable variables of actor also be trained when training critic network? if the truth is "yes". I think that is against the training illustration in the initial article。 thanks for your answer。

**Comment by linyufeishi** [comment]
hello,morvan,
Thanks again for your answer,your code about reforcement learning is the most elegant I have seen in the github, I have learned more from it ,Yeah,the new solution using tf.stop_gradient function can work well ,and I also solve the problem by specifying the  var_list for the function like the following:

self.training_paramters = tf.get_collection(tf.GraphKeys.TRAINABLE_VARIABLES,scope='Critic/eval_net')
self.train_op = tf.train.AdamOptimizer(self.lr).minimize(self.loss,var_list=self.training_params)

**Comment by MorvanZhou** [comment]
Excellent! I have fixed this problem and repost the code. Thanks for your suggestion.
